### ISSUE TITLE
Implementation detail of the robust loss

### ISSUE THREAD

**Issue by dfrumkin** [intel-isl/MiDaS#73]
Looking at https://gist.github.com/ranftlr/1d6194db2e1dffa0a50c9b0a9549cbd2#file-loss-py-L31, it looks like you are computing the median over all the values, including invalid ones.  Shouldn't this be a masked median?  (It is not available off the shelf, but can be implemented efficiently in a roundabout way).  And if it's a bug / feature, then how does it relate to the trimming portion of the loss?

**Comment by Mofafa** [comment]
I am wondering if the part 
`trimmed, _ = torch.sort(res.view(-1), descending=False)[
        : int(len(res) * (1.0 - trim))
    ]
`

should be changed to
`
trimmed, _ = torch.sort(res.view(-1), descending=False)
trimmed = trimmed[: int(len(res) * (1.0 - trim))]
`

**Comment by liuxiao1468** [comment]
> I am wondering if the part
> `trimmed, _ = torch.sort(res.view(-1), descending=False)[ : int(len(res) * (1.0 - trim)) ] `
> 
> should be changed to
> `trimmed, _ = torch.sort(res.view(-1), descending=False)`
> `trimmed = trimmed[: int(len(res.view(-1)) * (1.0 - trim))]`

I think it should be changed to the second one, the first line won't perform trimming.

**Comment by mathmax12** [comment]
@dfrumkin  Hi Guys,
I wonder are you able to reproduce the training script. I am pretty to this and wonder how you achieved this.
Thanks.

**Comment by CJCHEN1230** [comment]
Same problem.

**Comment by chhkang** [comment]
Asumming why should we use median in this loss? Just calculating mean on masked values makes loss performance bad?
